src/main/java/HttpServer.java

```diff
@@ -11,7 +11,8 @@ public static void main(String[] args) throws IOException {
     SocketIO socket = new SocketIO(arguments.getPortNumber());
     HttpReaderWriter httpReaderWriter = new HttpReaderWriter(socket, builder);
     FileIO fileIO = new FileIO(arguments.getDirectory());
-    HttpHandlerSelector handler = new HttpHandlerSelector(fileIO);
+    HandleGetRequest getRequest = new HandleGetRequest(fileIO);
+    HttpHandlerSelector handler = new HttpHandlerSelector(getRequest);
```

Comment:
What you're saying makes sense, especially when the time comes to make this server multithreaded. 
I will have to do some research on this either sometime this evening or tomorrow morning. Might be something to ask about at standup.